src/Microsoft.SqlTools.ServiceLayer/Profiler/IProfilerSessionMonitor.cs

```diff
@@ -22,7 +22,7 @@ public interface IProfilerSessionMonitor
         /// <summary>
         /// Starts monitoring a profiler session
         /// </summary>
-        bool StartMonitoringSession(string viewerId, IXEventSession session);
+        bool StartMonitoringStream(string viewerId, IXEventSession session, ConnectionInfo connInfo);
```

Comment:
we pass the connection to `XEventSessionFactory.GetXEventSession` to create the session.  It seems strange to also pass it here.  I think the `GetXEventSession` should do everything we need with connection and once we start the session we're already connected.  Also, I don't see a good reason to rename the method.  XEvents uses the session terminology and so I think it makes sense to stick with that, unless there is a good reason to change the name.

Reply:
It is passed there because we need to get the connectionString from the connectionInfo for XElite, and the ProfilerSession object does not include that by default. Regardless of how its done there must be a way to get the proper connection string in there.